Panic in bencode

```
panic: interface conversion: interface {} is nil, not string [recovered]
	panic: interface conversion: interface {} is nil, not string

goroutine 504 [running]:
github.com/anacrolix/torrent/bencode.(*Decoder).Decode.func1(0xc0004df610)
	/home/dev/go/src/github.com/anacrolix/torrent/bencode/decode.go:33 +0x114
panic(0xa4b2e0, 0xc000152420)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/anacrolix/torrent/tracker.(*Peer).fromDictInterface(0xc0004df228, 0xc000152360)
	/home/dev/go/src/github.com/anacrolix/torrent/tracker/peer.go:18 +0x1ea
github.com/anacrolix/torrent/tracker.(*Peers).UnmarshalBencode(0xc0004060f0, 0xc00021e500, 0x68, 0x280, 0xc0004060f0, 0xc0004060f0)
	/home/dev/go/src/github.com/anacrolix/torrent/tracker/http.go:57 +0x1ba
github.com/anacrolix/torrent/bencode.(*Decoder).parseUnmarshaler(0xc0001c6400, 0xa29080, 0xc0004060f0, 0x197, 0x30)
	/home/dev/go/src/github.com/anacrolix/torrent/bencode/decode.go:458 +0x18a
github.com/anacrolix/torrent/bencode.(*Decoder).parseValue(0xc0001c6400, 0xa29080, 0xc0004060f0, 0x197, 0x5, 0xa29080, 0xc0004060f0)
	/home/dev/go/src/github.com/anacrolix/torrent/bencode/decode.go:478 +0xb1
github.com/anacrolix/torrent/bencode.(*Decoder).parseDict(0xc0001c6400, 0xaacbe0, 0xc0004060c0, 0x199, 0xc0004df500, 0x41cc99)
	/home/dev/go/src/github.com/anacrolix/torrent/bencode/decode.go:322 +0x1e4
github.com/anacrolix/torrent/bencode.(*Decoder).parseValue(0xc0001c6400, 0xaacbe0, 0xc0004060c0, 0x199, 0xc0004060c0, 0x199, 0x100c00045c428)
	/home/dev/go/src/github.com/anacrolix/torrent/bencode/decode.go:499 +0x2ec
github.com/anacrolix/torrent/bencode.(*Decoder).Decode(0xc0001c6400, 0x9f7de0, 0xc0004060c0, 0x0, 0x0)
	/home/dev/go/src/github.com/anacrolix/torrent/bencode/decode.go:46 +0x201
github.com/anacrolix/torrent/bencode.Unmarshal(0xc000546000, 0xb1, 0x200, 0x9f7de0, 0xc0004060c0, 0xd7, 0x0)
	/home/dev/go/src/github.com/anacrolix/torrent/bencode/api.go:136 +0xd0
github.com/anacrolix/torrent/tracker.announceHTTP(0xc000800c40, 0x3c, 0x3b10cb0e97ffe685, 0x21299ecba052d70a, 0x3042502d08f804f9, 0xbbc4b1482d343230, 0xabaf34d488ded999, 0x0, 0x17d5fc1c, 0x0, ...)
	/home/dev/go/src/github.com/anacrolix/torrent/tracker/http.go:131 +0x6e3
github.com/anacrolix/torrent/tracker.Announce.Do(0xc000800c40, 0x3c, 0x3b10cb0e97ffe685, 0x21299ecba052d70a, 0x3042502d08f804f9, 0xbbc4b1482d343230, 0xabaf34d488ded999, 0x0, 0x17d5fc1c, 0x0, ...)
	/home/dev/go/src/github.com/anacrolix/torrent/tracker/tracker.go:75 +0x1b3
github.com/anacrolix/torrent.(*trackerScraper).announce(0xc00053dd40, 0x0, 0x0, 0x0, 0x45d964b800, 0xbf244993ddeb098f, 0x211f3177d, 0x11d3320)
	/home/dev/go/src/github.com/anacrolix/torrent/tracker_scraper.go:122 +0x489
github.com/anacrolix/torrent.(*trackerScraper).Run(0xc00053dd40)
	/home/dev/go/src/github.com/anacrolix/torrent/tracker_scraper.go:135 +0x2a1
created by github.com/anacrolix/torrent.(*Torrent).startScrapingTracker
	/home/dev/go/src/github.com/anacrolix/torrent/torrent.go:1281 +0x450
```

Content of resp.Body buffer from tracker response (IPs were redacted):

```
d8:completei0e10:incompletei0e8:intervali720e12:min intervali100e5:peersld2:ip11:93.3.50.1334:porti34345eed2:ip13:46.222.188.334:porti38625eed2:ip13:90.78.229.1204:porti8999eeee
```

It works on other torrent clients without issues.

--- a/tracker/peer.go
+++ b/tracker/peer.go
@@ -15,7 +15,9 @@ type Peer struct {
 // Set from the non-compact form in BEP 3.
 func (p *Peer) fromDictInterface(d map[string]interface{}) {
 	p.IP = net.ParseIP(d["ip"].(string))
-	p.ID = []byte(d["peer id"].(string))
+	if _, ok := d["peer id"]; ok {
+		p.ID = []byte(d["peer id"].(string))
+	}
 	p.Port = int(d["port"].(int64))
 }
 

